deepchem/models/torch_models/ferminet.py

```diff
@@ -22,9 +24,55 @@ def test_f(x: np.ndarray) -> np.ndarray:
     return 2 * np.log(np.random.uniform(low=0, high=1.0, size=np.shape(x)[0]))
 
 
-class Ferminet:
+class Ferminet(torch.nn.Module):
+    """Approximates the log probability of the wave function of a molecule system using DNNs.
+  """
+
+    def __init__(self,
+                 nucleon_pos: torch.Tensor,
+                 nuclear_charge: torch.Tensor,
+                 spin: tuple,
+                 inter_atom: torch.Tensor,
+                 n_one: List = [256, 256, 256, 256],
+                 n_two: List = [32, 32, 32, 32],
+                 determinant: int = 16) -> None:
+        """
+    Parameters:
+    -----------
+    nucleon_pos: torch.Tensor
+        Torch tensor containing nucleus information of the molecule
+    nuclear_charge: torch.Tensor
+        Torch tensor containing the number of electron for each atom in the molecule
+    spin: tuple
+        Tuple in the format of (up_spin, down_spin)
+    inter_atom: torch.Tensor
+        Torch tensor containing the pairwise distances between the atoms in the molecule
+    n_one: List
+      List of hidden units for the one-electron stream in each layer
+    n_two: List
+      List of hidden units for the two-electron stream in each layer
+    determinant: int
+      Number of determinants for the final solution
+    """
+        super(Ferminet, self).__init__()
+        if len(n_one) != len(n_two):
+            raise ValueError(
+                "The number of layers in one-electron and two-electron stream should be equal"
+            )
+        else:
+            self.layers = len(n_one)
+        self.nucleon_pos = nucleon_pos
+        self.determinant = determinant
+        self.spin = spin
+        self.inter_atom = inter_atom
+        self.n_one = n_one
+        self.n_two = n_two
+        self.determinant = determinant
+
+
+class FerminetModel(TorchModel):
```

Comment:
Can you add this class to the docs?

Reply:
Can I do this once the model is fully completed? As I am breaking the chunk of PR into smaller ones and am adding one function by function.